### PR TITLE
Remove LogEnqueueResult heap allocation in AsyncLogger Enqueue

### LINKED ISSUE #274: LogEnqueueResult in AsyncLogger Enqueue causes heap allocation
The [LogEnqueueResult local function](https://github.com/luigiberrettini/NLog.Targets.Syslog/blob/v6.0.2/src/NLog.Targets.Syslog/AsyncLogger.cs#L119-L124) captures variables in the enclosing scope, [therefore](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions#variable-capture) it is implemented as a delegate type and causes heap allocations.

### PR DESCRIPTION
**Pull request by snakefoot**
Closes #274

### CODE REVIEW

File: src/NLog.Targets.Syslog/AsyncLogger.cs [hunk 1]
```diff
@@ -116,20 +116,15 @@ private Task ProcessQueueAsync(MessageBuilder messageBuilder, TaskCompletionSour
...
-                InternalLogger.Debug("[Syslog] {0} '{1}'", message, asyncLogEventInfo.ToFormattedMessage());
-            }
+            bool success = queue.TryAdd(asyncLogEventInfo, timeout, token);
 
-            if (queue.TryAdd(asyncLogEventInfo, timeout, token))
+            if (InternalLogger.IsDebugEnabled)
             {
-                LogEnqueueResult("Enqueued");
-                return;
+                InternalLogger.Debug("[Syslog] {0} '{1}'", success ? "Enqueued" : "Failed Enqueue", asyncLogEventInfo.ToFormattedMessage());
```
Comment (luigiberrettini) [inline src/NLog.Targets.Syslog/AsyncLogger.cs:123]: Nitpick: `Failed Enqueue` to `Failed enqueuing`

File: src/NLog.Targets.Syslog/AsyncLogger.cs [hunk 2]
```diff
@@ -116,20 +116,15 @@ private Task ProcessQueueAsync(MessageBuilder messageBuilder, TaskCompletionSour
 
         private void Enqueue(AsyncLogEventInfo asyncLogEventInfo, int timeout)
         {
-            void LogEnqueueResult(string message)
-            {
-                if (!InternalLogger.IsDebugEnabled)
-                    return;
-                InternalLogger.Debug("[Syslog] {0} '{1}'", message, asyncLogEventInfo.ToFormattedMessage());
-            }
+            bool success = queue.TryAdd(asyncLogEventInfo, timeout, token);
```
Comment (luigiberrettini) [inline src/NLog.Targets.Syslog/AsyncLogger.cs:119]: I would rename `success` with `enqueued`

File: src/NLog.Targets.Syslog/AsyncLogger.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog.Targets.Syslog/AsyncLogger.cs]: Changed

File: src/NLog.Targets.Syslog/AsyncLogger.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog.Targets.Syslog/AsyncLogger.cs]: Changed

**Pull request merged**